### PR TITLE
usdUtils: Added MissingReferenceValidator

### PR DESCRIPTION
**Pull request by beersandrew**
### Description of Change(s)
- created implementation, test, as well as associated entries in plugInfo.json & validatorTokens.h

### Fixes Issue(s)
- https://github.com/orgs/PixarAnimationStudios/projects/3/views/1?pane=issue&itemId=65424971

### Checklist

[X] I have created this PR based on the dev branch

[X] I have followed the [coding conventions](https://openusd.org/release/api/_page__coding__guidelines.html)

[X] I have added unit tests that exercise this functionality (Reference: 
[testing guidelines](https://openusd.org/release/api/_page__testing__guidelines.html))

[X] I have verified that all unit tests pass with the proposed changes

[X] I have submitted a signed Contributor License Agreement (Reference: 
[Contributor License Agreement instructions](https://openusd.org/release/contributing_to_usd.html#contributor-license-agreement))

### CODE REVIEW

File: pxr/usd/usdUtils/validators.cpp [hunk 1]
```diff
@@ -104,9 +104,44 @@ _PackageEncapsulationValidator(const UsdStagePtr& usdStage) {
...
+            UsdValidationErrorType::Error,
+            UsdValidationErrorSites {
+                UsdValidationErrorSite(
+                    rootLayer, SdfPath(unresolvedPath))
+            },
+            TfStringPrintf(
+            ("Found unresolvable external dependency "
+                "'%s'."), unresolvedPath.c_str())
+        );
+    }
```
Comment (beersandrew) [inline pxr/usd/usdUtils/validators.cpp:133]: @tallytalwar One note here, it looks like `complianceChecker.py` is doing another check [here](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usdUtils/complianceChecker.py#L181) which gets some diagnostic information, but it seems like this process of collecting diagnostic information should be started before opening the stage ([here](https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usdUtils/complianceChecker.py#L1086)). I tried doing this inside the validator with stage.Reload() but was unable to populate the diagnostics this way.

In my specific test, the errors were similar so I wanted your opinion on if it's possible to do something similar here, or if it's necessary to move the diagnostic part over in this case.

**Comment by jesschimein** [comment]
Filed as internal issue #USD-10405

**Comment by beersandrew** [comment]
Closing this PR, this work is now done on https://github.com/PixarAnimationStudios/OpenUSD/pull/3450

**Pull request closed without merging**